setup-homebrew/post.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -euo pipefail
+
+DEBUG="${1}"
+
+if [[ "${DEBUG}" == 'true' ]]; then
+  set -x
+fi
+
+if [[ -n "${STATE_TAP_SYMLINK-}" ]]; then
+  rm "${STATE_TAP_SYMLINK}"
+  mkdir "${STATE_TAP_SYMLINK}"
+  (shopt -s dotglob; mv "${GITHUB_WORKSPACE}/*" "${STATE_TAP_SYMLINK}" )
+  echo "Reset tap symlink."
+fi
+
+if [[ -n "${STATE_TOKEN_SET-}" ]]; then
+  git config --global --unset-all "http.${GITHUB_SERVER_URL}/.extraheader"
+  echo "Removed token."
+fi
+
+if [[ -n "${STATE_SETFACL_DIRECTORIES-}" ]]; then
+  (IFS=:; sudo setfacl -Rb ${STATE_SETFACL_DIRECTORIES})
+  echo "Reset permissions."
+fi
```

Comment:
This is a cleanup step to restore everything back to how it was before the job started.

* Tokens were previously never cleaned up. This meant subsequent jobs could read them. I consider this to be a security vulnerability.
* Symlinks should be reverted otherwise if the same runnner is used for two taps then we might have symlinks for two taps pointing to the same place.
* `setfacl` reverting I think is good to do. Probably won't _break_ if not done, but I still think it's good to clean up.

Reply:
Ok, I see, so this is for cleanup after all of the job has completed and not just `setup-homebrew`.

Good to me, then. Again: would like a lot more inline comments explaining what's going on to future readers.